src/test/java/org/springframework/data/elasticsearch/core/MappingBuilderTests.java

```diff
@@ -409,6 +412,41 @@ public void shouldUseFieldNameOnMultiField() throws IOException, JSONException {
 		assertEquals(expected, mapping, false);
 	}
 
+	@Test // DATAES-639
+	public void shouldUseIgnoreAbove(){
+		// given
+		elasticsearchTemplate.createIndex(IgnoreAboveEntity.class);
+		elasticsearchTemplate.putMapping(IgnoreAboveEntity.class);
+
+		// when
+		Map mapping = elasticsearchTemplate.getMapping(IgnoreAboveEntity.class);
+		Map properties = (Map) mapping.get("properties");
+		Map message = (Map) properties.get("message");
+
+		// then
+		java.lang.reflect.Field field = ReflectionUtils.findField(IgnoreAboveEntity.class, "message");
```

Comment:
no need to test if the testclass, which is defined 10 lines later, has this annotation and no need to assert the attribute of the annotation.

This test is about the `MappingBuilder`, so I would not store the mapping in the index, and retrieve it as map and then parse it. This would integration testing. I know, that a lot of tests in this class do this, but the question to answer is _does the MappingBuilder create the correct mapping_? For this you don't need to store the mapping in the index. Have a look at what `shouldBuildMappingsForGeoPoint()` does: Test the JSON string returned from the MappingBuilder to make sure that it contains the correct mapping.